README.md

```diff
@@ -3,104 +3,134 @@
+   health of other systems, but again it doesn't know about datasets
+   that follow periodic patterns. It is worth mentioning that Greenish
+   provides an endpoint to export metrics to Prometheus.
 
-**Airflow** knows about periods, but it is not a monitoring tool. Airflow can
-alert when a run fails, but if a computed data was mistakenly deleted Airflow
-stays unaware.
+*  **Airflow** knows about periods, but it is not a monitoring
+   tool. Airflow can alert when a run fails, but if an existing dataset
+   gets deleted accidentally, Airflow stays unaware.
 
-What set's Greenish apart is that, it knows about periods, and keeps checking
-the existence of the datasets.
+What sets Greenish apart is that it knows about periods, and keeps checking
+datasets for existence.
 
 ## Monitoring Jobs
 
-As mentioned earlier, monitoring jobs can be any executable program, as long as:
+As mentioned earlier, monitoring scripts are stand-alone programs,
+written in any language, that respect the following contract:
+
+* The scripts must be executable.
 
-- They are executable
-- They accept a variable number of `period` arguments, as their last argument
-  sets: `$ monitor_my_job.sh staging 2020-20-06-10 2020-20-06-11 ...`
-  The above is an example of a monitoring script that is written in a
-  `shell-like` scripting language. In the example, the script takes a
-  parameter for the environment (in this case, `staging` is applied), and takes
-  a set of `period` parameters, in the example: `2020-02-06-10` and
-  `2020-02-06-11` are passed.
+* The scripts must accept an arbitrary number of `period` arguments at
+  the end of their parameter list; e.g., for a script named
+  `monitor-foo`, running on the `staging` environment, asked to check
+  the status of three hourly periods:
 
-  The `check-command` entry for the above example will be:
+  ```shell
+  monitor-foo staging 2020-20-06-10 2020-20-06-11 2020-20-06-12
   ```
-    check-command: "monitor_my_job staging"
+
+  The `check-command` entry for the example above could be:
+
+  ```yaml
+    check-command: "monitor-foo staging"
     period-pattern: "yyyy-MM-dd-HH"
   ```
 
-- They print the health for every provided periods, exactly once, in the
-  following format: `greenish-period\t$PERIOD\t1` when the period's health is
-  OK, or `greenish-period\t$PERIOD\t0` otherwise. It is important that:
-    - `$PERIOD` is in the list of the provided periods.
-    - The text above is in a separate line, namely the line should match:
-      `^greenish-period\t.*\t(0|1)$`.
-    - The three parts of the line are tab separated.
-- The scripts can have any number of debugging/application output lines.
-- The script should exit with 0, under normal circumstances even if the entire
-  set of periods are not in a good health.
+- The scripts must print one diagnostic line per provided period in
+  one of the following two formats, where `1` indicates a successful
```

Comment:
It is actually one format, with different values for the exit code?

Reply:
I debated what was the best way to express this, and tried to make it obvious that the part that varies to signify success vs failure is the `0` or `1` at the end. What do you think? Would it be best if I had a single line that said `<...> (0|1)`, and a separate explanation for what `0` and `1` mean?

Reply:
Actually, you are right...